pkg/gui/controllers/scroll_off_margin.go

```diff
@@ -0,0 +1,68 @@
+package controllers
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	"github.com/jesseduffield/lazygit/pkg/utils"
+)
+
+// To be called after pressing up-arrow; checks whether the cursor entered the
+// top scroll-off margin, and so the view needs to be scrolled up one line
+func checkScrollUp(view types.IViewTrait, scrollOffMargin int, lineIdxBefore int, lineIdxAfter int) {
+	viewPortStart, viewPortHeight := view.ViewPortYBounds()
+
+	if linesToScroll := calculateLinesToScrollUp(
```

Comment:
```suggestion
	linesToScroll := calculateLinesToScrollUp(
		viewPortStart, viewPortHeight, scrollOffMargin, lineIdxBefore, lineIdxAfter,
	)
	if linesToScroll != 0 {
		view.ScrollUp(linesToScroll)
	}
```
I struggle to read multi-line if statements

Reply:
390cb2aa1f59